broker/lib/cache_policy_manager.py

```diff
@@ -0,0 +1,29 @@
+from broker.extensions import config
+
+
+class CachePolicyManager:
+    def __init__(self, cloudfront):
+        self.policies = {}
+        self.cloudfront = cloudfront
+
+    def get_managed_cache_policies(self) -> dict[str, str]:
+        policy_type = "managed"
+        if policy_type not in self.policies:
+            self._list_cache_policies(policy_type)
+        return self.policies[policy_type]
```

Comment:
if I'm reading the intent here correctly, we want `policies` to be a lazy-loaded property, and we're testing that it's loaded by looking for `"managed"` in the dict because we expect it to always be there. If that's the case, we can do this a little more idiomatically with something like:

```python
def CachePolicyManager:
    def __init__(self, cloudfront):
        self._policies = None
        self.cloudfront = cloudfront

    @property
    def policies(self) -> dict[str, str]:
        if self._policies is None:
            self._policies = self._list_cache_policies()
        return self._policies
```

(although now that I've typed that all out, I see that we're only ever getting `"managed"` policies, so I think I'm missing something here)

Reply:
I implemented the feedback: 6960f98d2286aaa9d3c31040a71bdb8775918d48. Thoughts?

You're right that currently we're only handling `managed` policies, but I'm future-proofing a little bit in case when we want to have this same code handle `custom` policies

Reply:
ah, yeah, that looks a lot better :+1: